lib/timers.js

```diff
@@ -128,23 +128,28 @@ function insert(item, unrefed) {
   var list = lists[msecs];
   if (!list) {
     debug('no %d list was found in insert, creating a new one', msecs);
-    // Make a new linked list of timers, and create a TimerWrap to schedule
-    // processing for the list.
-    list = new TimersList(msecs, unrefed);
-    L.init(list);
-    list._timer._list = list;
-
-    if (unrefed === true) list._timer.unref();
-    list._timer.start(msecs);
-
-    lists[msecs] = list;
-    list._timer[kOnTimeout] = listOnTimeout;
+    lists[msecs] = list = createTimersList(msecs, unrefed);
   }
 
   L.append(list, item);
   assert(!L.isEmpty(list)); // list is not empty
 }
 
+function createTimersList(msecs, unrefed) {
```

Comment:
The main reason is to keep the `insert()` size down, but it does also help a bit with readability I suppose.